src/Sulu/Component/CustomUrl/Routing/CustomUrlRouteProvider.php

```diff
@@ -180,4 +189,24 @@ private function decodePathInfo($pathInfo)
     {
         return rawurldecode($pathInfo);
     }
+
+    /**
+     * @param RouteCollection $collection
+     */
+    public function setRouteCollection(RouteCollection $collection)
```

Comment:
I don't think it is a good idea to introduce this method, if you only need it for tests... Then I would even prefer to set it via reflections in the test. If we introduce this method people might start using it for whatever reason...